src/SourcemapToolkit.SourcemapParser/MappingEntry.cs

```diff
@@ -2,46 +2,81 @@
 
 namespace SourcemapToolkit.SourcemapParser
 {
-	public class MappingEntry
+	public struct MappingEntry
 	{
 		/// <summary>
 		/// The location of the line of code in the transformed code
 		/// </summary>
-		public SourcePosition GeneratedSourcePosition;
+		public readonly SourcePosition GeneratedSourcePosition;
 
 		/// <summary>
 		/// The location of the code in the original source code
 		/// </summary>
-		public SourcePosition OriginalSourcePosition;
+		public readonly SourcePosition OriginalSourcePosition;
 
 		/// <summary>
 		/// The original name of the code referenced by this mapping entry
 		/// </summary>
-		public string OriginalName;
+		public readonly string OriginalName;
 
 		/// <summary>
 		/// The name of the file that originally contained this code
 		/// </summary>
-		public string OriginalFileName;
+		public readonly string OriginalFileName;
```

Comment:
I assume this would be something like `webpack://some/very/long/path/to/your/source/file.ts`, and we store this for every entry for the source map, which would mean we could easily be storing this 100+ times for a single file.

If that's true, would it be worth creating a single map of shortFileId -> OriginalFileName and just storing a shortFileId here?

Reply:
I think the string instance is already being shared for all strings in the SourceMap.  So what we're paying in overhead is the object pointer versus a smaller numeric value.  I agree that we can save space by moving to the smaller numeric value, but I think this review is already getting pretty large and if we pursue it it should be done separately.  (That, and I only allocated a few hours to making these improvements for the time being so I didn't have time to get to it.)